### PR TITLE
Add support for fprof and cprof profilers

### PR DESCRIPTION
**Pull request by djschlegel2**
Add support for fprof and cprof internal profiling

Bugzid 71356

### CODE REVIEW

File: src/basho_bench_profiler.erl [hunk 1]
```diff
@@ -0,0 +1,65 @@
+-module(basho_bench_profiler).
+
+-export([ maybe_start_profiler/1, maybe_terminate_profiler/1
```
Comment (ksnavely) [inline src/basho_bench_profiler.erl]: This spacing seems a bit goofy...should be fine all on one line w/o the leading space?

File: src/basho_bench_profiler.erl [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        ]).
+
+-include("basho_bench.hrl").
+
+maybe_start_profiler(Profiler) ->
+
+    case Profiler of 
+        false ->
+            {ok, 0};
+
```
Comment (ksnavely) [inline src/basho_bench_profiler.erl]: I think typically we do not include newlines in case statements

**Comment by ksnavely** [comment]
Whitespace comments but LGTM if working.

File: src/basho_bench_profiler.erl [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+            {ok, Pid};
+
+        fprof ->
+            FprofTraceFile = filename:join(basho_bench:get_test_dir(), "fprofTrace.log"),
+            ?CONSOLE("Starting fprof profiling to ~p\n", [FprofTraceFile]),
+            {ok, Pid} = fprof:start(),
+            fprof:trace([start, {file, FprofTraceFile}]),
+            {ok, Pid}
+
+    end.
```
Comment (chewbranca) [inline src/basho_bench_profiler.erl]: No need to use a case statement at all, you can do the pattern matching in the function clauses.

**Comment by chewbranca** [comment]
+1

**Pull request merged**